src/reader/parser/mod.rs

```diff
@@ -470,10 +470,13 @@ impl PullParser {
 
         // check and fix accumulated attributes prefixes
         for attr in attributes.iter_mut() {
-            match self.nst.get(attr.name.borrow().prefix_repr()) {
-                Some("") => attr.name.namespace = None,  // default namespace
-                Some(ns) => attr.name.namespace = Some(ns.into()),
-                None => return Some(self_error!(self; "Attribute {} prefix is unbound", attr.name))
+            for pfx in attr.name.prefix.iter() {
```

Comment:
`prefix` is an `Option<String>`; it is unidiomatic to use iterators API to access option contents. `if let Some(ref pfx)` would be more appropriate.